src/repository.ts

```diff
@@ -842,8 +842,12 @@ export class Repository implements IRemoteRepository {
   }
 
   public async commitFiles(message: string, files: any[]) {
-    return this.run(Operation.Commit, () =>
-      this.repository.commitFiles(message, files)
+    const args = { message: message, files: files };
+    return this.run(
+      Operation.Commit,
+      (obj: any = { message: message, files: files }) =>
+        this.repository.commitFiles(obj.message, obj.files),
+      args
     );
   }
```

Comment:
This is an example of how we are supposed to allow operations to have arguments overwritten by pre-operation hooks. Instead of an empty arrow function, we create one taking an object as an argument. This object holds a reference to the original arguments, but allow a new object to be passed overwriting the original arguments.